instrumentation/aws-sdk/aws-sdk-2.2/testing/src/main/groovy/io/opentelemetry/instrumentation/awssdk/v2_2/AbstractAws2SqsTracingTest.groovy

```diff
@@ -204,4 +245,132 @@ abstract class AbstractAws2SqsTracingTest extends InstrumentationSpecification {
     resp.messages().size() == 1
     assertSqsTraces()
   }
+
+  def "batch sqs producer-consumer services: sync"() {
+    setup:
+    def builder = SqsClient.builder()
+    configureSdkClient(builder)
+    def client = builder.build()
+
+    client.createQueue(createQueueRequest)
+
+    when:
+    client.sendMessageBatch(sendMessageBatchRequest)
+
+    def resp = client.receiveMessage(receiveMessageBatchRequest)
+    def totalAttrs = resp.messages().sum {it.messageAttributes().size() }
+
+    then:
+    resp.messages().size() == 3
+    sqsAttributeInjectionEnabled || totalAttrs == 18 // No additional attributes
+    !sqsAttributeInjectionEnabled || totalAttrs == 18 + 2 // Once not injected due to too many attrs
```

Comment:
Nit: can you use `if else` here? it'll be slightly easier to read that way

Reply:
I was not sure if Spock would only verify top-level expressions here, since we are not using explicit assert.

Reply:
Ah, right, you'd have to use `assert` there. How about using `&&` then? It might be my own bias, but I think it's a bit easier to parse than `||` in this case

Reply:
This reads much nicer and should hopefully work equivalently: 9be32c2

Reply:
Yeah, that is great. Thanks!